lib/chef/resource/launchd.rb

```diff
@@ -0,0 +1,285 @@
+#
+# Author:: Mike Dodge (<mikedodge04@gmail.com>)
+# Copyright:: Copyright (c) 2015 Facebook, Inc.
+# License:: Apache License, Version 2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+require 'chef/resource/file'
+require 'chef/provider/launchd'
+require 'chef/mixin/securable'
+
+class Chef
+  class Resource
+    class Launchd < Chef::Resource::File
+      include Chef::Mixin::Securable
+
+      provides :launchd
```

Comment:
This shouldn't be necessary since you're not providing an `os` constraint, this happens when you declare the resource_name.

https://github.com/chef/chef/blob/master/lib/chef/resource.rb#L941